diff --git mix.exs mix.exs
index f9849a7..e416e08 100644
--- mix.exs
+++ mix.exs
@@ -9,8 +9,8 @@ defmodule SampleApp.Mixfile do
      compilers: [:phoenix, :gettext] ++ Mix.compilers,
      build_embedded: Mix.env == :prod,
      start_permanent: Mix.env == :prod,
-     aliases: aliases,
-     deps: deps]
+     aliases: aliases(),
+     deps: deps()]
   end
 
   # Configuration for the OTP application.
@@ -30,12 +30,13 @@ defmodule SampleApp.Mixfile do
   #
   # Type `mix help deps` for examples and options.
   defp deps do
-    [{:phoenix, "~> 1.1.5"},
+    [{:phoenix, "~> 1.1.9"},
      {:postgrex, ">= 0.0.0"},
-     {:phoenix_ecto, "~> 2.0"},
-     {:phoenix_html, "~> 2.4"},
+     {:plug, "~> 1.1.0"},
+     {:phoenix_ecto, "~> 2.0.1"},
+     {:phoenix_html, "~> 2.6.0"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
-     {:gettext, "~> 0.9"},
+     {:gettext, "~> 0.9.0"},
      {:cowboy, "~> 1.0"}]
   end
 
