diff --git brunch-config.js brunch-config.js
index fa906bd..f2e7eec 100644
--- brunch-config.js
+++ brunch-config.js
@@ -59,7 +59,8 @@ exports.config = {
 
   modules: {
     autoRequire: {
-      "js/app.js": ["web/static/js/app"]
+      "js/app.js": ["web/static/js/app"],
+      "js\\app.js": ["web/static/js/app"]
     }
   },
 
diff --git mix.exs mix.exs
index 945c62b..9d125eb 100644
--- mix.exs
+++ mix.exs
@@ -30,7 +30,7 @@ defmodule SampleApp.Mixfile do
   #
   # Type `mix help deps` for examples and options.
   defp deps do
-    [{:phoenix, "~> 1.1.1"},
+    [{:phoenix, "~> 1.1.0"},
      {:phoenix_ecto, "~> 2.0"},
      {:postgrex, ">= 0.0.0"},
      {:phoenix_html, "~> 2.3"},
diff --git priv/gettext/en/LC_MESSAGES/errors.po priv/gettext/en/LC_MESSAGES/errors.po
index 02b2a76..9e03f00 100644
--- priv/gettext/en/LC_MESSAGES/errors.po
+++ priv/gettext/en/LC_MESSAGES/errors.po
@@ -10,10 +10,6 @@ msgstr ""
 msgid "can't be blank"
 msgstr ""
 
-## From Ecto.Changeset.unique_constraint/3
-msgid "has already been taken"
-msgstr ""
-
 ## From Ecto.Changeset.put_change/3
 msgid "is invalid"
 msgstr ""
diff --git priv/gettext/errors.pot priv/gettext/errors.pot
index 893d26c..b1adbad 100644
--- priv/gettext/errors.pot
+++ priv/gettext/errors.pot
@@ -7,10 +7,6 @@
 msgid "can't be blank"
 msgstr ""
 
-## From Ecto.Changeset.unique_constraint/3
-msgid "has already been taken"
-msgstr ""
-
 ## From Ecto.Changeset.put_change/3
 msgid "is invalid"
 msgstr ""
diff --git test/test_helper.exs test/test_helper.exs
index be885a8..d3effc2 100644
--- test/test_helper.exs
+++ test/test_helper.exs
@@ -1,6 +1,6 @@
 ExUnit.start
 
-Mix.Task.run "ecto.create", ~w(-r SampleApp.Repo --quiet)
-Mix.Task.run "ecto.migrate", ~w(-r SampleApp.Repo --quiet)
+Mix.Task.run "ecto.create", ["--quiet"]
+Mix.Task.run "ecto.migrate", ["--quiet"]
 Ecto.Adapters.SQL.begin_test_transaction(SampleApp.Repo)
 
