diff --git mix.exs mix.exs
index e416e08..7e101fd 100644
--- mix.exs
+++ mix.exs
@@ -9,8 +9,8 @@ defmodule SampleApp.Mixfile do
      compilers: [:phoenix, :gettext] ++ Mix.compilers,
      build_embedded: Mix.env == :prod,
      start_permanent: Mix.env == :prod,
-     aliases: aliases(),
-     deps: deps()]
+     aliases: aliases,
+     deps: deps]
   end
 
   # Configuration for the OTP application.
@@ -30,13 +30,12 @@ defmodule SampleApp.Mixfile do
   #
   # Type `mix help deps` for examples and options.
   defp deps do
-    [{:phoenix, "~> 1.1.9"},
+    [{:phoenix, "~> 1.1.3"},
+     {:phoenix_ecto, "~> 2.0"},
      {:postgrex, ">= 0.0.0"},
-     {:plug, "~> 1.1.0"},
-     {:phoenix_ecto, "~> 2.0.1"},
-     {:phoenix_html, "~> 2.6.0"},
+     {:phoenix_html, "~> 2.3"},
      {:phoenix_live_reload, "~> 1.0", only: :dev},
-     {:gettext, "~> 0.9.0"},
+     {:gettext, "~> 0.9"},
      {:cowboy, "~> 1.0"}]
   end
 
diff --git package.json package.json
index 150178b..69da608 100644
--- package.json
+++ package.json
@@ -2,12 +2,12 @@
   "repository": {
   },
   "dependencies": {
-    "babel-brunch": "~6.0.0",
-    "brunch": "~2.1.3",
-    "clean-css-brunch": "~1.8.0",
-    "css-brunch": "~1.7.0",
-    "javascript-brunch": "~1.8.0",
-    "uglify-js-brunch": "~1.7.0",
+    "babel-brunch": "^6.0.0",
+    "brunch": "^2.1.3",
+    "clean-css-brunch": ">= 1.0 < 1.8",
+    "css-brunch": ">= 1.0 < 1.8",
+    "javascript-brunch": ">= 1.0 < 1.8",
+    "uglify-js-brunch": ">= 1.0 < 1.8",
     "phoenix": "file:deps/phoenix",
     "phoenix_html": "file:deps/phoenix_html"
   }
