diff --git config/prod.exs config/prod.exs
index 920b334..00dbb52 100644
--- config/prod.exs
+++ config/prod.exs
@@ -55,10 +55,6 @@ config :logger, level: :info
 #
 #     config :sample_app, SampleApp.Endpoint, server: true
 #
-# You will also need to set the application root to `.` in order
-# for the new static assets to be served after a hot upgrade:
-#
-#     config :sample_app, SampleApp.Endpoint, root: "."
 
 # Finally import the config/prod.secret.exs
 # which should be versioned separately.
diff --git mix.exs mix.exs
index 112b641..6cdd934 100644
--- mix.exs
+++ mix.exs
@@ -30,7 +30,7 @@ defmodule SampleApp.Mixfile do
   #
   # Type `mix help deps` for examples and options.
   defp deps do
-    [{:phoenix, "~> 1.2.0"},
+    [{:phoenix, "~> 1.2.4"},
      {:phoenix_pubsub, "~> 1.0"},
      {:phoenix_ecto, "~> 3.0"},
      {:postgrex, ">= 0.0.0"},
