diff --git config/config.exs config/config.exs
index 49827c9..47bdd68 100644
--- config/config.exs
+++ config/config.exs
@@ -6,16 +6,14 @@
 use Mix.Config
 
 # General application configuration
-config :sample_app,
-  ecto_repos: [SampleApp.Repo]
+config :sample_app, ecto_repos: [SampleApp.Repo]
 
 # Configures the endpoint
 config :sample_app, SampleAppWeb.Endpoint,
   url: [host: "localhost"],
   secret_key_base: "aaaaaaaa",
   render_errors: [view: SampleAppWeb.ErrorView, accepts: ~w(html json)],
-  pubsub: [name: SampleApp.PubSub,
-           adapter: Phoenix.PubSub.PG2]
+  pubsub: [name: SampleApp.PubSub, adapter: Phoenix.PubSub.PG2]
 
 # Configures Elixir's Logger
 config :logger, :console,
@@ -24,4 +22,4 @@ config :logger, :console,
 
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
-import_config "#{Mix.env}.exs"
+import_config "#{Mix.env()}.exs"
diff --git lib/sample_app_web/endpoint.ex lib/sample_app_web/endpoint.ex
index 58ce45a..8613087 100644
--- lib/sample_app_web/endpoint.ex
+++ lib/sample_app_web/endpoint.ex
@@ -8,7 +8,9 @@ defmodule SampleAppWeb.Endpoint do
   # You should set gzip to true if you are running phoenix.digest
   # when deploying your static files in production.
   plug Plug.Static,
-    at: "/", from: :sample_app, gzip: false,
+    at: "/",
+    from: :sample_app,
+    gzip: false,
     only: ~w(css fonts images js favicon.ico robots.txt)
 
   # Code reloading can be explicitly enabled under the
diff --git mix.exs mix.exs
index 3c379b7..4494a36 100644
--- mix.exs
+++ mix.exs
@@ -33,7 +33,7 @@ defmodule SampleApp.Mixfile do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:phoenix, "~> 1.3.3"},
+      {:phoenix, "~> 1.3.1"},
       {:phoenix_pubsub, "~> 1.0"},
       {:phoenix_ecto, "~> 3.2"},
       {:postgrex, ">= 0.0.0"},
