diff --git mix.exs mix.exs
index 9269afe..7d50cf5 100644
--- mix.exs
+++ mix.exs
@@ -33,7 +33,7 @@ defmodule SampleApp.MixProject do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:phoenix, "~> 1.4.5"},
+      {:phoenix, "~> 1.4.6"},
       {:phoenix_pubsub, "~> 1.1"},
       {:phoenix_ecto, "~> 4.0"},
       {:ecto_sql, "~> 3.0"},
diff --git test/support/data_case.ex test/support/data_case.ex
index 7ad0f52..ef81ce3 100644
--- test/support/data_case.ex
+++ test/support/data_case.ex
@@ -46,7 +46,7 @@ defmodule SampleApp.DataCase do
   def errors_on(changeset) do
     Ecto.Changeset.traverse_errors(changeset, fn {message, opts} ->
       Regex.replace(~r"%{(\w+)}", message, fn _, key ->
-        opts |> Map.get(String.to_existing_atom(key), key) |> to_string()
+        opts |> Keyword.get(String.to_existing_atom(key), key) |> to_string()
       end)
     end)
   end
