diff --git .formatter.exs .formatter.exs
index 738a390..b443429 100644
--- .formatter.exs
+++ .formatter.exs
@@ -1,5 +1,5 @@
 [
-  import_deps: [:ecto, :phoenix],
+  import_deps: [:phoenix],
   inputs: ["*.{ex,exs}", "{config,lib,priv,test}/**/*.{ex,exs}"],
   subdirectories: ["priv/*/migrations"]
 ]
diff --git mix.exs mix.exs
index 377558d..c45d7eb 100644
--- mix.exs
+++ mix.exs
@@ -35,8 +35,9 @@ defmodule SampleApp.MixProject do
     [
       {:phoenix, "~> 1.4.0-rc"},
       {:phoenix_pubsub, "~> 1.1"},
-      {:phoenix_ecto, "~> 4.0"},
-      {:ecto_sql, "~> 3.0-rc"},
+      {:phoenix_ecto, "~> 3.5"},
+      {:ecto, "~> 3.0-rc", override: true},
+      {:ecto_sql, "~> 3.0-rc", override: true},
       {:postgrex, ">= 0.0.0-rc"},
       {:phoenix_html, "~> 2.11"},
       {:phoenix_live_reload, "~> 1.2-rc", only: :dev},
