diff --git .formatter.exs .formatter.exs
index b443429..738a390 100644
--- .formatter.exs
+++ .formatter.exs
@@ -1,5 +1,5 @@
 [
-  import_deps: [:phoenix],
+  import_deps: [:ecto, :phoenix],
   inputs: ["*.{ex,exs}", "{config,lib,priv,test}/**/*.{ex,exs}"],
   subdirectories: ["priv/*/migrations"]
 ]
diff --git mix.exs mix.exs
index c45d7eb..377558d 100644
--- mix.exs
+++ mix.exs
@@ -35,9 +35,8 @@ defmodule SampleApp.MixProject do
     [
       {:phoenix, "~> 1.4.0-rc"},
       {:phoenix_pubsub, "~> 1.1"},
-      {:phoenix_ecto, "~> 3.5"},
-      {:ecto, "~> 3.0-rc", override: true},
-      {:ecto_sql, "~> 3.0-rc", override: true},
+      {:phoenix_ecto, "~> 4.0"},
+      {:ecto_sql, "~> 3.0-rc"},
       {:postgrex, ">= 0.0.0-rc"},
       {:phoenix_html, "~> 2.11"},
       {:phoenix_live_reload, "~> 1.2-rc", only: :dev},
