diff --git .gitignore .gitignore
index f190c9b..7a97fd7 100644
--- .gitignore
+++ .gitignore
@@ -32,3 +32,11 @@ npm-debug.log
 # we ignore priv/static. You may want to comment
 # this depending on your deployment strategy.
 /priv/static/
+
+# Files matching config/*.secret.exs pattern contain sensitive
+# data and you should not commit them into version control.
+#
+# Alternatively, you may comment the line below and commit the
+# secrets files as long as you replace their contents by environment
+# variables.
+/config/*.secret.exs
diff --git assets/package.json assets/package.json
index a4ba0e7..c81fa4f 100644
--- assets/package.json
+++ assets/package.json
@@ -14,7 +14,7 @@
     "@babel/preset-env": "^7.0.0",
     "babel-loader": "^8.0.0",
     "copy-webpack-plugin": "^4.5.0",
-    "css-loader": "^2.1.1",
+    "css-loader": "^0.28.10",
     "mini-css-extract-plugin": "^0.4.0",
     "optimize-css-assets-webpack-plugin": "^4.0.0",
     "uglifyjs-webpack-plugin": "^1.2.4",
diff --git config/dev.exs config/dev.exs
index 217905f..14ea2bf 100644
--- config/dev.exs
+++ config/dev.exs
@@ -1,14 +1,5 @@
 use Mix.Config
 
-# Configure your database
-config :sample_app, SampleApp.Repo,
-  username: "postgres",
-  password: "postgres",
-  database: "sample_app_dev",
-  hostname: "localhost",
-  show_sensitive_data_on_connection_error: true,
-  pool_size: 10
-
 # For development, we disable any cache and enable
 # debugging and code reloading.
 #
@@ -58,10 +49,10 @@ config :sample_app, SampleAppWeb.Endpoint,
 config :sample_app, SampleAppWeb.Endpoint,
   live_reload: [
     patterns: [
-      ~r"priv/static/.*(js|css|png|jpeg|jpg|gif|svg)$",
-      ~r"priv/gettext/.*(po)$",
-      ~r"lib/sample_app_web/{live,views}/.*(ex)$",
-      ~r"lib/sample_app_web/templates/.*(eex)$"
+      ~r{priv/static/.*(js|css|png|jpeg|jpg|gif|svg)$},
+      ~r{priv/gettext/.*(po)$},
+      ~r{lib/sample_app_web/views/.*(ex)$},
+      ~r{lib/sample_app_web/templates/.*(eex)$}
     ]
   ]
 
@@ -74,3 +65,11 @@ config :phoenix, :stacktrace_depth, 20
 
 # Initialize plugs at runtime for faster development compilation
 config :phoenix, :plug_init_mode, :runtime
+
+# Configure your database
+config :sample_app, SampleApp.Repo,
+  username: "postgres",
+  password: "postgres",
+  database: "sample_app_dev",
+  hostname: "localhost",
+  pool_size: 10
diff --git config/prod.exs config/prod.exs
index ead09c8..0d797f4 100644
--- config/prod.exs
+++ config/prod.exs
@@ -10,6 +10,7 @@ use Mix.Config
 # which you should run after static files are built and
 # before starting your production server.
 config :sample_app, SampleAppWeb.Endpoint,
+  http: [:inet6, port: System.get_env("PORT") || 4000],
   url: [host: "example.com", port: 80],
   cache_static_manifest: "priv/static/cache_manifest.json"
 
@@ -50,16 +51,21 @@ config :logger, level: :info
 #
 # Check `Plug.SSL` for all available options in `force_ssl`.
 
-# ## Using releases (Elixir v1.9+)
+# ## Using releases (distillery)
 #
 # If you are doing OTP releases, you need to instruct Phoenix
-# to start each relevant endpoint:
+# to start the server for all endpoints:
+#
+#     config :phoenix, :serve_endpoints, true
+#
+# Alternatively, you can configure exactly which server to
+# start per endpoint:
 #
 #     config :sample_app, SampleAppWeb.Endpoint, server: true
 #
-# Then you can assemble a release by calling `mix release`.
-# See `mix help release` for more information.
+# Note you can't rely on `System.get_env/1` when using releases.
+# See the releases documentation accordingly.
 
-# Finally import the config/prod.secret.exs which loads secrets
-# and configuration from environment variables.
+# Finally import the config/prod.secret.exs which should be versioned
+# separately.
 import_config "prod.secret.exs"
diff --git config/prod.secret.exs config/prod.secret.exs
index dc5b6f0..a62a8db 100644
--- config/prod.secret.exs
+++ config/prod.secret.exs
@@ -1,29 +1,19 @@
-# In this file, we load production configuration and
-# secrets from environment variables. You can also
-# hardcode secrets, although such is generally not
-# recommended and you have to remember to add this
-# file to your .gitignore.
 use Mix.Config
 
-database_url =
-  System.get_env("DATABASE_URL") ||
-    raise """
-    environment variable DATABASE_URL is missing.
-    For example: ecto://USER:PASS@HOST/DATABASE
-    """
-
-config :sample_app, SampleApp.Repo,
-  # ssl: true,
-  url: database_url,
-  pool_size: String.to_integer(System.get_env("POOL_SIZE") || "10")
-
-secret_key_base =
-  System.get_env("SECRET_KEY_BASE") ||
-    raise """
-    environment variable SECRET_KEY_BASE is missing.
-    You can generate one by calling: mix phx.gen.secret
-    """
-
+# In this file, we keep production configuration that
+# you'll likely want to automate and keep away from
+# your version control system.
+#
+# You should document the content of this
+# file or create a script for recreating it, since it's
+# kept out of version control and might be hard to recover
+# or recreate for your teammates (or yourself later on).
 config :sample_app, SampleAppWeb.Endpoint,
-  http: [:inet6, port: String.to_integer(System.get_env("PORT") || "4000")],
   secret_key_base: "aaaaaaaa"
+
+# Configure your database
+config :sample_app, SampleApp.Repo,
+  username: "postgres",
+  password: "postgres",
+  database: "sample_app_prod",
+  pool_size: 15
diff --git config/test.exs config/test.exs
index 31fdfee..166e50f 100644
--- config/test.exs
+++ config/test.exs
@@ -1,13 +1,5 @@
 use Mix.Config
 
-# Configure your database
-config :sample_app, SampleApp.Repo,
-  username: "postgres",
-  password: "postgres",
-  database: "sample_app_test",
-  hostname: "localhost",
-  pool: Ecto.Adapters.SQL.Sandbox
-
 # We don't run a server during test. If one is required,
 # you can enable the server option below.
 config :sample_app, SampleAppWeb.Endpoint,
@@ -16,3 +8,11 @@ config :sample_app, SampleAppWeb.Endpoint,
 
 # Print only warnings and errors during test
 config :logger, level: :warn
+
+# Configure your database
+config :sample_app, SampleApp.Repo,
+  username: "postgres",
+  password: "postgres",
+  database: "sample_app_test",
+  hostname: "localhost",
+  pool: Ecto.Adapters.SQL.Sandbox
diff --git lib/sample_app_web/templates/page/index.html.eex lib/sample_app_web/templates/page/index.html.eex
index 8cbd9d8..0784473 100644
--- lib/sample_app_web/templates/page/index.html.eex
+++ lib/sample_app_web/templates/page/index.html.eex
@@ -1,6 +1,6 @@
 <section class="phx-hero">
   <h1><%= gettext "Welcome to %{name}!", name: "Phoenix" %></h1>
-  <p>A productive web framework that<br/>does not compromise speed or maintainability.</p>
+  <p>A productive web framework that<br/>does not compromise speed and maintainability.</p>
 </section>
 
 <section class="row">
diff --git mix.exs mix.exs
index 7d50cf5..bec62bb 100644
--- mix.exs
+++ mix.exs
@@ -33,7 +33,7 @@ defmodule SampleApp.MixProject do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:phoenix, "~> 1.4.6"},
+      {:phoenix, "~> 1.4.1"},
       {:phoenix_pubsub, "~> 1.1"},
       {:phoenix_ecto, "~> 4.0"},
       {:ecto_sql, "~> 3.0"},
diff --git test/support/data_case.ex test/support/data_case.ex
index ef81ce3..a71a146 100644
--- test/support/data_case.ex
+++ test/support/data_case.ex
@@ -45,8 +45,8 @@ defmodule SampleApp.DataCase do
   """
   def errors_on(changeset) do
     Ecto.Changeset.traverse_errors(changeset, fn {message, opts} ->
-      Regex.replace(~r"%{(\w+)}", message, fn _, key ->
-        opts |> Keyword.get(String.to_existing_atom(key), key) |> to_string()
+      Enum.reduce(opts, message, fn {key, value}, acc ->
+        String.replace(acc, "%{#{key}}", to_string(value))
       end)
     end)
   end
