diff --git assets/css/app.css assets/css/app.scss
similarity index 100%
rename from data/sample-app/1.5.0-rc.0/assets/css/app.css
rename to data/sample-app/1.5.0/assets/css/app.scss
diff --git assets/js/app.js assets/js/app.js
index 4195b56..dc8033b 100644
--- assets/js/app.js
+++ assets/js/app.js
@@ -1,7 +1,7 @@
 // We need to import the CSS so that webpack will load it.
 // The MiniCssExtractPlugin is used to separate it out into
 // its own CSS file.
-import "../css/app.css"
+import "../css/app.scss"
 
 // webpack automatically bundles all modules in your
 // entry points. Those entry points can be configured
diff --git assets/package.json assets/package.json
index 2edc95d..8f7026f 100644
--- assets/package.json
+++ assets/package.json
@@ -18,6 +18,8 @@
     "babel-loader": "^8.0.0",
     "copy-webpack-plugin": "^5.1.1",
     "css-loader": "^3.4.2",
+    "sass-loader": "^8.0.2",
+    "node-sass": "^4.13.1",
     "mini-css-extract-plugin": "^0.9.0",
     "optimize-css-assets-webpack-plugin": "^5.0.1",
     "terser-webpack-plugin": "^2.3.2",
diff --git assets/webpack.config.js assets/webpack.config.js
index 1be6e21..dd77c3d 100644
--- assets/webpack.config.js
+++ assets/webpack.config.js
@@ -34,16 +34,12 @@ module.exports = (env, options) => {
           }
         },
         {
-          test: /\.css$/,
+          test: /\.[s]?css$/,
           use: [
             MiniCssExtractPlugin.loader,
-            {
-              loader: 'css-loader',
-              options: {
-                sourceMap: devMode
-              }
-            }
-          ]
+            'css-loader',
+            'sass-loader',
+          ],
         }
       ]
     },
diff --git lib/sample_app_web.ex lib/sample_app_web.ex
index 78548ec..6b5bfd8 100644
--- lib/sample_app_web.ex
+++ lib/sample_app_web.ex
@@ -80,9 +80,12 @@ defmodule SampleAppWeb do
       # Use all HTML functionality (forms, tags, etc)
       use Phoenix.HTML
 
-      # Import convenience functions for LiveView rendering
+      # Import LiveView helpers (live_render, live_component, live_patch, etc)
       import Phoenix.LiveView.Helpers
 
+      # Import basic rendering functionality (render, render_layout, etc)
+      import Phoenix.View
+
       import SampleAppWeb.ErrorHelpers
       import SampleAppWeb.Gettext
       alias SampleAppWeb.Router.Helpers, as: Routes
diff --git mix.exs mix.exs
index 52ea030..ad341a8 100644
--- mix.exs
+++ mix.exs
@@ -33,7 +33,7 @@ defmodule SampleApp.MixProject do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:phoenix, "~> 1.5.0-rc.0", override: true},
+      {:phoenix, "~> 1.5.0"},
       {:phoenix_ecto, "~> 4.1"},
       {:ecto_sql, "~> 3.4"},
       {:postgrex, ">= 0.0.0"},
@@ -41,7 +41,7 @@ defmodule SampleApp.MixProject do
       {:floki, ">= 0.0.0", only: :test},
       {:phoenix_html, "~> 2.11"},
       {:phoenix_live_reload, "~> 1.2", only: :dev},
-      {:phoenix_live_dashboard, "~> 0.1.0"},
+      {:phoenix_live_dashboard, "~> 0.2.0"},
       {:telemetry_metrics, "~> 0.4"},
       {:telemetry_poller, "~> 0.4"},
       {:gettext, "~> 0.11"},
